browser/ui/webui/settings/brave_settings_import_data_handler_base.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/webui/settings/brave_settings_import_data_handler_base.h"
+
+#include "brave/browser/importer/brave_importer_p3a.h"
+
+namespace settings {
+
+BraveImportDataHandlerBase::BraveImportDataHandlerBase() = default;
+BraveImportDataHandlerBase::~BraveImportDataHandlerBase() = default;
+
+void BraveImportDataHandlerBase::StartImport(
+    const importer::SourceProfile& source_profile,
+    uint16_t imported_items) {
+  ImportDataHandler::StartImport(source_profile, imported_items);
+
+  type_ = source_profile.importer_type;
+}
+
+void BraveImportDataHandlerBase::ImportEnded() {
+  ImportDataHandler::ImportEnded();
+
+  if (!import_did_succeed_)
```

Comment:
@iefremov should we always report regardless of import result?

Reply:
I think it's not v important given that probably import should be mostly successful. But we can verify it with @davidtemkin . In general, I think we should just preserve the original behaviour